### PR TITLE
builtin bit_count

### PR DESCRIPTION
**Pull request by cuiyuan**
builtin bit_count function

### CODE REVIEW

File: expression/builtin_other.go [hunk 1]
```diff
@@ -263,5 +264,25 @@ type builtinBitCountSig struct {
...
+	args, err := b.evalArgs(row)
+	if err != nil {
+		return types.Datum{}, errors.Trace(err)
+	}
+	arg := args[0]
+	if arg.IsNull() {
+		return d, nil
+	}
+	sc := new(variable.StatementContext)
+	sc.IgnoreTruncate = true
```
Comment (hanfei1991) [inline expression/builtin_other.go]: mysql> select bit_count("111111111111111111111111111834982309");
+---------------------------------------------------+
| bit_count("111111111111111111111111111834982309") |
+---------------------------------------------------+
|                                                64 |
+---------------------------------------------------+
1 row in set, 1 warning (0.00 sec)

Comment (hanfei1991) [inline expression/builtin_other.go]: If exceed int64, it returns 64.

File: expression/builtin_other.go [hunk 2]
```diff
@@ -263,5 +264,25 @@ type builtinBitCountSig struct {
 
 // See https://dev.mysql.com/doc/refman/5.7/en/bit-functions.html#function_bit-count
 func (b *builtinBitCountSig) eval(row []types.Datum) (d types.Datum, err error) {
-	return d, errFunctionNotExists.GenByArgs("BIT_COUNT")
+	args, err := b.evalArgs(row)
+	if err != nil {
+		return types.Datum{}, errors.Trace(err)
```
Comment (shenli) [inline expression/builtin_other.go]: return d, errors.Trace(err)

File: expression/builtin_other_test.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
+package expression
```
Comment (shenli) [inline expression/builtin_other_test.go]: Miss license.

File: expression/builtin_other_test.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+		in := types.NewDatum(test.origin)
+		f, _ := fc.getFunction(datumsToConstants([]types.Datum{in}), s.ctx)
+		count, err := f.eval(nil)
+		c.Assert(err, IsNil)
+		sc := new(variable.StatementContext)
+		sc.IgnoreTruncate = true
+		res, err := count.ToInt64(sc)
+		c.Assert(err, IsNil)
+		c.Assert(res, Equals, test.count)
+	}
```
Comment (shenli) [inline expression/builtin_other_test.go]: Please also add negative test cases.

File: expression/builtin_other.go [hunk 5]
```diff
@@ -263,5 +264,29 @@ type builtinBitCountSig struct {
...
+	}
+	arg := args[0]
+	if arg.IsNull() {
+		return d, nil
+	}
+	sc := new(variable.StatementContext)
+	sc.IgnoreTruncate = true
+	bin, err := arg.ToInt64(sc)
+	if err != nil {
+		if strings.Contains(err.Error(), "overflows") {
```
Comment (hanfei1991) [inline expression/builtin_other.go]: use terror.ErrorEqual

**Comment by hanfei1991** [comment]
LGTM

**Pull request merged**